tests/unit/fem/test_var_order.cpp

```diff
@@ -696,7 +696,14 @@ static void TestSolveVec(FiniteElementSpace &fespace)
 
    GridFunction x(&fespace);
    x = 0.0;
-   x.ProjectBdrCoefficient(exsol, ess_attr);
+   if (x.BdrVectorDim() == 1)
+   {
+      x.ProjectBdrCoefficientNormal(exsol, ess_attr);
+   }
+   else
+   {
+      x.ProjectBdrCoefficientTangent(exsol, ess_attr);
+   }
```

Comment:
This is a good catch. Considering that this type of mistake is very easy to make, I think we should update `ProjectBdrCoefficient` to handle H(div) and H(curl) spaces and dispatch to `ProjectBdrCoefficientNormal` and `ProjectBdrCoefficientTangent` when needed.

Reply:
Well, that is for discussion. `ProjectBdrCoefficient()` makes sense even for H(div) or H(curl) if you know what you are doing (so the dimension of the vector coefficient is matching). But I agree that most of the times people do it accidentally. Now they will know, because it will give them an error 🚫 . Alternatively, there could be that auto-magic you suggest 🪄 .